### DISCUSSION TITLE
CRUD bulk filtering

### DISCUSSION THREAD

**Discussion by giovanni-motterle-mia-care** [mia-platform/community#66] [Q&A]
Hi, I noted that crud bulk operations are not consistent in their interface.
Examples:

- PATCH /v2/*/bulk requires to use _sk in the filter object of the payload if you want to modify objects only in certain states (\_\_STATE\_\_ is not allowed) and **only** the first object that match the filter is updated
- POST /v2/*/state requires to use \_\_STATE\_\_ and _sk is not allowed in the filter object of the payload. **All** objects that match the filter are affected.

Why do these differences exist? Am I missing something?

**Comment by fredmaggiowski** [comment]
Hi, isn't `_sk` for skipping documents? Maybe you're referring to `_st`? 

Consider that the parameters starting with `_` are the special query parameters provided by the crud to interact with the property it defines; __STATE__ is one of those and as such is provided with the `_st` param and not with a `__STATE__` query param.

**Reply by giovanni-motterle-mia-care** [reply]
Sorry, I meant _st. In both cases they are not used as query parameter but in the body. If we require using _st and not \_\_STATE\_\_ in the filter property of one bulk endpoint, then it should be consistent in all the others bulk endpoints.
The same argument applies also for the number of document affected, if one bulk endpoint modifies all documents that match the filter, then all bulk endpoints should work in this way
